### PR TITLE
Agent work (WIP)

### PR DESCRIPTION
**Pull request by abist**
Not to be merged yet. Added step implementation to the job page. Here's an example - 

steps

Related PR in Sql Tools Service here - https://github.com/Microsoft/sqltoolsservice/pull/597

### CODE REVIEW

**Comment by kburtram** [comment]
@abist could you please rebase this PR off `master`.  We should start working out of there for SQL Agent changes since the initial payload has already been merged.

File: extensions-modules/lib/configurations/config.d.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { IConfig } from '../languageservice/interfaces';
+export default class Config implements IConfig {
```
Comment (kburtram) [inline extensions-modules/lib/configurations/config.d.ts]: why are we checking in the `lib` directory?  Isn't this build output?

File: src/sql/parts/dashboard/dashboard.module.ts [hunk 2]
```diff
@@ -51,7 +51,7 @@ import { JobHistoryComponent } from 'sql/parts/jobManagement/views/jobHistory.co
 let baseComponents = [DashboardHomeContainer, DashboardComponent, DashboardWidgetWrapper, DashboardWebviewContainer,
 					DashboardWidgetContainer, DashboardGridContainer, DashboardErrorContainer, DashboardNavSection, WebviewContent, WidgetContent,
 					ComponentHostDirective, BreadcrumbComponent, ControlHostContent, DashboardControlHostContainer,
-					JobsViewComponent, AgentViewComponent, JobHistoryComponent];
+					JobsViewComponent, AgentViewComponent, JobHistoryComponent, StepsViewComponent];
```
Comment (kburtram) [inline src/sql/parts/dashboard/dashboard.module.ts]: please name this `JobStepsViewComponent`.  I'd like to keep `Job` prefix on these related classes.

File: src/sql/parts/jobManagement/views/jobHistory.component.ts [hunk 3]
```diff
@@ -44,6 +47,9 @@ export class JobHistoryComponent extends Disposable implements OnInit, OnDestroy
 
 	private prevJobId: string = undefined;
 	private isVisible: boolean = false;
+	private stepRows: StepsViewRow[] = [];
+	private _showSteps: boolean = false;
```
Comment (kburtram) [inline src/sql/parts/jobManagement/views/jobHistory.component.ts]: there is inconsistent usage of `_` prefix on private variable.  Can you please make these usage consistent in this file?

File: src/sql/parts/jobManagement/views/stepsView.component.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+import { Tree } from 'vs/base/parts/tree/browser/treeImpl';
+import { IInstantiationService } from 'vs/platform/instantiation/common/instantiation';
+import { Disposable, IDisposable } from 'vs/base/common/lifecycle';
+import { IBootstrapService, BOOTSTRAP_SERVICE_ID } from 'sql/services/bootstrap/bootstrapService';
+import { IJobManagementService } from '../common/interfaces';
+import { DashboardServiceInterface } from 'sql/parts/dashboard/services/dashboardServiceInterface.service';
+import { AgentJobHistoryInfo } from 'sqlops';
+import { StepsViewController, StepsViewDataSource, StepsViewFilter, StepsViewRenderer, StepsViewRow, StepsViewModel} from 'sql/parts/jobManagement/views/stepsViewTree';
+
+export const DASHBOARD_SELECTOR: string = 'stepsview-component';
```
Comment (kburtram) [inline src/sql/parts/jobManagement/views/stepsView.component.ts]: may `JOBSTEPSVIEW_SELECTOR` is a better name

File: src/sql/parts/jobManagement/views/stepsView.component.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+			this._treeDataSource.data = this.stepRows;
+			if (!this._tree) {
+				this._tree = new Tree(this._tableContainer.nativeElement, {
+					controller: this._treeController,
+					dataSource: this._treeDataSource,
+					filter: this._treeFilter,
+					renderer: this._treeRenderer
+				});
+			}
+			this._tree.layout(1024);
```
Comment (kburtram) [inline src/sql/parts/jobManagement/views/stepsView.component.ts]: what is the number `1024` do?

File: src/sql/parts/jobManagement/views/stepsView.css [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+	margin-top: 4px;
+}
+
+.list-row .label {
+	padding-left: 10px;
+	display: flex;
+	text-align: center;
+}
+
+.passed {
```
Comment (kburtram) [inline src/sql/parts/jobManagement/views/stepsView.css]: so we provide more specific class names and\or provide better scoping of the styles?  Some of these seem very likely to collide at some point.

File: src/sql/parts/jobManagement/views/stepsView.component.ts (hunk 5 above)
Comment (abist) [reply src/sql/parts/jobManagement/views/stepsView.component.ts]: That's the standard webpage size

File: src/sql/parts/jobManagement/views/jobStepsView.component.ts [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+import { DashboardServiceInterface } from 'sql/parts/dashboard/services/dashboardServiceInterface.service';
+import { AgentJobHistoryInfo } from 'sqlops';
+import { JobStepsViewController, JobStepsViewDataSource, JobStepsViewFilter,
+	JobStepsViewRenderer, JobStepsViewRow, JobStepsViewModel} from 'sql/parts/jobManagement/views/jobStepsViewTree';
+
+export const JOBSTEPSVIEW_SELECTOR: string = 'jobstepsview-component';
+
+@Component({
+	selector: JOBSTEPSVIEW_SELECTOR,
+	templateUrl: decodeURI(require.toUrl('./stepsView.component.html'))
```
Comment (kburtram) [inline src/sql/parts/jobManagement/views/jobStepsView.component.ts]: can you update the HTML file name to match the component file name, `jobStepsView...`

File: src/sql/parts/jobManagement/views/jobStepsView.component.ts [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+		@Inject(forwardRef(() => ChangeDetectorRef)) private _cd: ChangeDetectorRef,
+		@Inject(forwardRef(() => DashboardServiceInterface)) private _dashboardService: DashboardServiceInterface
+	) {
+		super();
+		this._jobManagementService = bootstrapService.jobManagementService;
+	}
+
+	ngOnInit() {
+		let ownerUri: string = this._dashboardService.connectionManagementService.connectionInfo.ownerUri;
+		const self = this;
```
Comment (kburtram) [inline src/sql/parts/jobManagement/views/jobStepsView.component.ts]: does look like we use `self` in this function.

File: src/sql/parts/jobManagement/views/jobStepsView.component.ts [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+			this._treeDataSource.data = this.stepRows;
+			if (!this._tree) {
+				this._tree = new Tree(this._tableContainer.nativeElement, {
+					controller: this._treeController,
+					dataSource: this._treeDataSource,
+					filter: this._treeFilter,
+					renderer: this._treeRenderer
+				});
+			}
+			this._tree.layout(1024);
```
Comment (kburtram) [inline src/sql/parts/jobManagement/views/jobStepsView.component.ts]: what is `1024` mean here?

Comment (abist) [reply src/sql/parts/jobManagement/views/jobStepsView.component.ts]: It's the default size of a webpage.

File: src/sql/parts/jobManagement/views/jobHistory.component.ts [hunk 10]
```diff
@@ -19,6 +19,9 @@ import { AgentViewComponent } from 'sql/parts/jobManagement/agent/agentView.comp
 import { JobHistoryController, JobHistoryDataSource,
 	JobHistoryRenderer, JobHistoryFilter, JobHistoryModel, JobHistoryRow } from 'sql/parts/jobManagement/views/jobHistoryTree';
 import { AgentJobHistoryInfo, AgentJobInfo } from 'sqlops';
+import { JobStepsViewComponent } from 'sql/parts/jobManagement/views/jobStepsView.component';
+import { JobStepsViewRow } from './jobStepsViewTree';
+import { OnChanges } from '~@angular/core/src/core';
```
Comment (anthonydresser) [inline src/sql/parts/jobManagement/views/jobHistory.component.ts]: This should be just from `from '@angular/core'`

**Pull request merged**